core/src/main/java/org/apache/brooklyn/feed/http/HttpFeed.java

```diff
@@ -331,52 +366,55 @@ protected void preStart() {
-            
-            getPoller().scheduleAtFixedRate(pollJob, new DelegatingPollHandler<HttpToolResponse>(handlers), minPeriod);
-        }
-    }
+            pollJob = new Callable<HttpToolResponse>() {
+                public HttpToolResponse call() throws Exception {
+                    if (log.isTraceEnabled()) log.trace("http polling for {} sensors at {}", entity, pollInfo);
 
-    // TODO Should we really trustAll for https? Make configurable?
-    private HttpClient createHttpClient(HttpPollIdentifier pollIdentifier) {
-        URI uri = pollIdentifier.uriProvider.get();
-        HttpClientBuilder builder = HttpTool.httpClientBuilder()
-                .trustAll()
-                .laxRedirect(true);
-        if (uri != null) builder.uri(uri);
-        if (uri != null) builder.credential(pollIdentifier.credentials);
-        if (pollIdentifier.connectionTimeout != null) {
-            builder.connectionTimeout(pollIdentifier.connectionTimeout);
-        }
-        if (pollIdentifier.socketTimeout != null) {
-            builder.socketTimeout(pollIdentifier.socketTimeout);
+                    UsernamePassword creds = null;
+                    if (pollInfo.credentials.isPresent()) {
+                        creds =  new UsernamePassword(
+                                pollInfo.credentials.get().getUserPrincipal().getName(),
+                                pollInfo.credentials.get().getPassword());
+                    }
+
+                    HttpResponse response =  pollInfo.httpExecutor.execute(new HttpRequest.Builder()
+                            .headers(pollInfo.headers)
+                            .uri(pollInfo.uriProvider.get())
+                            .credentials(creds)
+                            .method(pollInfo.method)
+                            .body(pollInfo.body)
+                            .config(HttpConfig.builder()
+                                    .trustSelfSigned(true)
+                                    .trustAll(true)
+                                    .laxRedirect(true)
+                                    .build())
+                            .build());
+                    return createHttpToolRespose(response);
+                }};
+                getPoller().scheduleAtFixedRate(pollJob, new DelegatingPollHandler(handlers), minPeriod);
         }
-        return builder.build();
     }
 
     @SuppressWarnings("unchecked")
     protected Poller<HttpToolResponse> getPoller() {
-        return (Poller<HttpToolResponse>) super.getPoller();
+        return  (Poller<HttpToolResponse>) super.getPoller();
+    }
+
+    @SuppressWarnings("unchecked")
+    private HttpToolResponse createHttpToolRespose(HttpResponse response) throws IOException {
+        int responseCode = response.code();
+
+        Map<String,? extends List<String>> headers = (Map<String, List<String>>) (Map<?, ?>) response.headers().asMap();
+
+        byte[] content = null;
+        if (response.getContentLength() > 0) {
```

Comment:
Better copy the current logic from [HttpToolResponse](https://github.com/apache/brooklyn-server/blob/master/utils/common/src/main/java/org/apache/brooklyn/util/http/HttpToolResponse.java#L69-L81) as it's been battle-tested.